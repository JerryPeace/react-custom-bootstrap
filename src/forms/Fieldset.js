import PropTypes from 'prop-types';
import React from 'react';
import { Basic } from './Basic';

class Fieldset extends Basic {
    static propTypes = {
        title: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    };

    render() {
        const {
            children,
            className,
            title
        } = this.props;
        return (
            <fieldset className={className}>
                <legend>{title}</legend>
                <div className='fieldset-content'>{children}</div>
            </fieldset>
        );
    }
}

export default Fieldset;
