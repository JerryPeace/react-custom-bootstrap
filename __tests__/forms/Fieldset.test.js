import React from 'react';
import { mount } from 'enzyme';
import toJSON from'enzyme-to-json';
import Fieldset from 'forms/Fieldset';

describe('Fieldset', () => {
    test('render fieldset correctly', () => {
        const fieldset = mount(
            <Fieldset title='test-title'>
                <div>test</div>
            </Fieldset>
        );
        expect(toJSON(fieldset)).toMatchSnapshot();
    });
});
