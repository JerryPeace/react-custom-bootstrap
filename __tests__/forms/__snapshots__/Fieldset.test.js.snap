// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset render fieldset correctly 1`] = `
<Fieldset
  disableDirtyCheck={false}
  title="test-title"
>
  <fieldset>
    <legend>
      test-title
    </legend>
    <div
      className="fieldset-content"
    >
      <div>
        test
      </div>
    </div>
  </fieldset>
</Fieldset>
`;
